c/dec/huffman.h

```diff
@@ -33,11 +33,60 @@ static const uint16_t kMaxHuffmanTableSize[] = {
 
 #define BROTLI_HUFFMAN_MAX_CODE_LENGTH_CODE_LENGTH 5
 
+#if !defined(BROTLI_HUFFMAN_CODE_FAST_LOAD)
 typedef struct {
   uint8_t bits;    /* number of bits used for this symbol */
   uint16_t value;  /* symbol value or table offset */
 } HuffmanCode;
 
+/* To allow optimization of HuffmanCode accesses in hot paths,
+ * please use the following macros to:
+ *
+ *  - prepare a HuffmanCode pointer for access
+ *     ( BROTLI_HC_PREPARE_TO_ACCESS(HuffmanCode*) )
+ *  - update the pointer each time it is changed
+ *     ( BROTLI_HC_UPDATE(HuffmanCode*) )
+ *  - get the bits or value fields
+ *     ( BROTLI_HC_GET_BITS/VALUE(HuffmanCode*) )
+ *
+ * For example, assuming |table| contains a HuffmanCode pointer:
+ *
+ *   BROTLI_HC_PREPARE_TO_ACCESS(table);
+ *   BROTLI_HC_UPDATE(table);
+ *   *bits = BROTLI_HC_GET_BITS(table);
+ *   *value = BROTLI_HC_GET_VALUE(table);
+ *   table += offset;
+ *   BROTLI_HC_UPDATE(table);
+ *   *bits2 = BROTLI_HC_GET_BITS(table);
+ *   *value2 = BROTLI_HC_GET_VALUE(table);
+ *
+ */
+
+#define BROTLI_HC_PREPARE_TO_ACCESS(H)
+#define BROTLI_HC_UPDATE(H)
+
+/* Getters must be given a pointer to a HuffmanCode. */
+#define BROTLI_HC_GET_BITS(H) (H->bits)
+#define BROTLI_HC_GET_VALUE(H) (H->value)
+
+/* Setters must be given a HuffmanCode. */
+#define BROTLI_HC_SET_BITS(H, B) H.bits = B
+#define BROTLI_HC_SET_VALUE(H, V) H.value = V
+#else
```

Comment:
"else" also should have this comment: /* BROTLI_HUFFMAN_CODE_FAST_LOAD */
perhaps a new line before and after else would improve readability.

Reply:
Done.